output/output.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/env"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/ziputil"
+	"github.com/bitrise-steplib/steps-xcode-test/testaddon"
+	"github.com/bitrise-steplib/steps-xcode-test/testartifact"
+)
+
+// Exporter ...
+type Exporter interface {
+	ExportXCResultBundle(deployDir, xcResultPath, scheme string)
+	ExportTestRunResult(failed bool)
+	ExportXcodebuildBuildLog(deployDir, xcodebuildBuildLog string)
+	ExportXcodebuildTestLog(deployDir, xcodebuildTestLog string)
+	ExportSimulatorDiagnostics(deployDir, pth, name string)
+	ExportUITestArtifacts(xcResultPath, scheme string)
+}
+
+type exporter struct {
+	envRepository        env.Repository
+	logger               log.Logger
+	testAddonExporter    testaddon.Exporter
+	testArtifactExporter testartifact.Exporter
+}
+
+// NewExporter ...
+func NewExporter(envRepository env.Repository, logger log.Logger, testAddonExporter testaddon.Exporter, testArtifactExporter testartifact.Exporter) Exporter {
+	return &exporter{
+		envRepository:        envRepository,
+		logger:               logger,
+		testAddonExporter:    testAddonExporter,
+		testArtifactExporter: testArtifactExporter,
+	}
+}
+
+func (e exporter) ExportTestRunResult(failed bool) {
+	status := "succeeded"
+	if failed {
+		status = "failed"
+	}
+	if err := e.envRepository.Set("BITRISE_XCODE_TEST_RESULT", status); err != nil {
+		e.logger.Warnf("Failed to export: BITRISE_XCODE_TEST_RESULT, error: %s", err)
+	}
+}
+
+func (e exporter) ExportXCResultBundle(deployDir, xcResultPath, scheme string) {
+	// export xcresult bundle
+	if err := e.envRepository.Set("BITRISE_XCRESULT_PATH", xcResultPath); err != nil {
+		e.logger.Warnf("Failed to export: BITRISE_XCRESULT_PATH, error: %s", err)
+	}
+
+	xcresultZipPath := filepath.Join(deployDir, filepath.Base(xcResultPath)+".zip")
+	if err := output.ZipAndExportOutput(xcResultPath, xcresultZipPath, "BITRISE_XCRESULT_ZIP_PATH"); err != nil {
+		e.logger.Warnf("Failed to export: BITRISE_XCRESULT_ZIP_PATH, error: %s", err)
+	}
+
+	// export xcresult for the testing addon
+	if addonResultPath := os.Getenv(configs.BitrisePerStepTestResultDirEnvKey); len(addonResultPath) > 0 {
+		e.logger.Println()
+		e.logger.Infof("Exporting test results")
+
+		if err := e.testAddonExporter.CopyAndSaveMetadata(testaddon.AddonCopy{
+			SourceTestOutputDir:   xcResultPath,
+			TargetAddonPath:       addonResultPath,
+			TargetAddonBundleName: scheme,
+		}); err != nil {
+			e.logger.Warnf("Failed to export test results, error: %s", err)
+		}
+	}
+}
+
+func (e exporter) ExportXcodebuildBuildLog(deployDir, xcodebuildBuildLog string) {
+	pth, err := saveRawOutputToLogFile(xcodebuildBuildLog)
+	if err != nil {
+		e.logger.Warnf("Failed to save the Raw Output, err: %s", err)
+	}
+
+	deployPth := filepath.Join(deployDir, "xcodebuild_build.log")
+	if err := command.CopyFile(pth, deployPth); err != nil {
+		// TODO: restore error handling
```

Comment:
What do these TODOs mean?

Reply:
With this PR all of the export errors are treated as a warning, previously there were some exceptions: https://github.com/bitrise-steplib/steps-xcode-test/blob/master/main.go#L520-L621

This TODO is for reconsidering if we should handle any issue as an error and make the step failing during output export.

Reply:
Do we plan to do these before the next release? Or we plan to release these as breaking changes?

Reply:
TODOs resolved by restoring the original error handling: https://github.com/bitrise-steplib/steps-xcode-test/pull/181/commits/54822c24c5eaa6686dab5d41ec62ad09be1a6df2